test/example/cpp-unit.cpp

```diff
@@ -538,6 +538,56 @@ void test_bits() {
   c0.value = Literal(int64_t(-1));
   assert_equal(getMaxBits(&u), 32);
 
+  u.type = Type::i32;
+  c0.type = Type::i32;
+
+  u.op = ExtendS8Int32;
+  c0.value = Literal(int8_t(0));
+  assert_equal(getMaxBits(&u), 0);
+  c0.value = Literal(int8_t(127));
+  assert_equal(getMaxBits(&u), 7);
+  c0.value = Literal(int8_t(128));
+  assert_equal(getMaxBits(&u), 32);
+
+  u.op = ExtendS16Int32;
+  c0.value = Literal(int16_t(0));
+  assert_equal(getMaxBits(&u), 0);
+  c0.value = Literal(int16_t(0x7FFF));
+  assert_equal(getMaxBits(&u), 15);
+  c0.value = Literal(int16_t(-1));
```

Comment:
```suggestion
  c0.value = Literal(int16_t(0x8000));
```
I think that is clearer given line 555. This way they are easy to compare.